### PR TITLE
gopaddle lite 4.2.5

### PR DESCRIPTION
**Pull request by renugadevi-2613**
### Thank you for making MicroK8s better

Please reference the issue this PR is fixing, or provide a description of the problem addressed.

*Also verify you have:*
* [X] Read the [contributions](https://github.com/ubuntu/microk8s/blob/master/CONTRIBUTING.md) page.
* [X] Submitted the [CLA form](https://ubuntu.com/legal/contributors/agreement), if you are a first time contributor.

### CODE REVIEW

File: addons/gopaddle-lite/enable [hunk 1]
```diff
@@ -0,0 +1,181 @@
+#!/usr/bin/env bash
+
+set -e
+
+source $SNAP/actions/common/utils.sh
+
+KUBECTL="$SNAP/microk8s-kubectl.wrapper"
+HELM="$SNAP/microk8s-helm3.wrapper"
+
+KUBE_VERSION=$($KUBECTL version | grep Server| cut -d '{' -f 2| cut -d ',' -f 2| cut -d ':' -f 2 |cut -d '+' -f 1|cut -d '"' -f 2)
```
Comment (neoaggelos) [inline addons/gopaddle-lite/enable:10]: How about:

```suggestion
# will return something like "26+"
KUBE_VERSION=$($KUBECTL version -o json | $SNAP/usr/bin/jq -r '.serverVersion.minor')
```

File: addons/gopaddle-lite/enable [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+
+CURRENT_DIR=$(cd $(dirname "${BASH_SOURCE[0]}") && pwd)
+
+STATIC_VERSION="24"
+if [ "$KUBE_VERSION" -gt $STATIC_VERSION ]
+then
+    echo "$KUBE_VERSION is not supported"
+fi
```
Comment (neoaggelos) [inline addons/gopaddle-lite/enable:18]: The `%+` part drops the extra `+` from the version string, if any.

```suggestion
if [ "${KUBE_VERSION%+}" -gt $STATIC_VERSION ]
then
    echo "WARNING: Detected Kubernetes version $KUBE_VERSION, which is not supported"
fi
```

**Review by neoaggelos — APPROVED** [review]
LGTM! Thanks a lot for this PR!

Left one minor comment for the kubernetes version detection logic, which would make maintenance easier in the future.

**Comment by neoaggelos** [comment]
Hi @renugadevi-2613, pinging for visibility, can you look at the review comments? We can then move this PR forward, thanks a lot!

**Comment by renugadevi-2613** [comment]
Hi @neoaggelos, I have made changes based on the suggestion and resolved the issues. Can we close the PR?

**Comment by neoaggelos** [comment]
Hi @renugadevi-2613, I don't see any new commits on the PR, I suppose you are fine with the code as-is?

**Comment by neoaggelos** [comment]
@renugadevi-2613 Can you rebase against the latest changes in main? Thanks!

**Comment by renugadevi-2613** [comment]
Hi @neoaggelos, Can I give a new PR after I rebase the changes?

**Comment by ktsakalozos** [comment]
Thank you for this PR @renugadevi-2613. There are a couple of more issues that I will address after merging it.

Each addon gets it own page in our docs liked from [1]. I have created a minimal page for gopaddle-lite in [2]. Please let me know if there is anything else to appear there.

[1] https://microk8s.io/docs/addons
[2] https://discuss.kubernetes.io/t/addon-gopaddle-lite/22120

**Comment by renugadevi-2613** [comment]
Hi @ktsakalozos, Thanks for accepting the PR. I will look forward to your suggestions on the addon. The minimal document looks good. Will it be possible to add some addional steps? Once the gopaddle-lite addon is enabled, the below command will wait until all the gopaddle services move to the Ready state.

``` microk8s.kubectl wait --for=condition=ready pod -l released-by=gopaddle -n gp-lite ```

Once the services are ready, the gopaddle dashboard can be accessed using the endpoint [http://<NODE_EXTERNAL-IP](http://external-ip/)>:30003. If no EXTERNAL_IP exists for the node, the INTERNAL_IP can be used. The below command can be used to get the node's EXTERNAL_IP or the INTERNAL_IP.

``` microk8s kubectl get nodes -o wide```

**Pull request merged**